src/cmake/thirdparty/FindLUA.cmake

```diff
@@ -20,36 +17,24 @@ if (NOT IS_DIRECTORY "${LUA_DIR}")
     message(FATAL_ERROR "Given LUA_DIR is not a directory: ${LUA_DIR}")
 endif()
 
-# Find includes directory
-find_path( LUA_INCLUDE_DIR lua.hpp
-           PATHS  ${LUA_DIR}/include/
-                  ${LUA_DIR}/include/lua
-           NO_DEFAULT_PATH
-           NO_CMAKE_ENVIRONMENT_PATH
-           NO_CMAKE_PATH
-           NO_SYSTEM_ENVIRONMENT_PATH
-           NO_CMAKE_SYSTEM_PATH)
-
-# Find libraries
-find_library( LUA_LIBRARY NAMES lua liblua
-              PATHS ${LUA_DIR}/lib
-              NO_DEFAULT_PATH
-              NO_CMAKE_ENVIRONMENT_PATH
-              NO_CMAKE_PATH
-              NO_SYSTEM_ENVIRONMENT_PATH
-              NO_CMAKE_SYSTEM_PATH)
+# Add LUA_DIR as env variable
+set(ENV{LUA_DIR} ${LUA_DIR})
 
+# Uncomment the following for more debug output in FindLUA
+#set(LUA_Debug TRUE)
 
-include(FindPackageHandleStandardArgs)
-# handle the QUIETLY and REQUIRED arguments and set LUA_FOUND to TRUE
-# if all listed variables are TRUE
-find_package_handle_standard_args(LUA  DEFAULT_MSG
-                                  LUA_INCLUDE_DIR
-                                  LUA_LIBRARY )
+# HACK: Workaround for lua@5.4 and older versions of cmake (e.g. 3.16)
+# which did not account for versions of lua beyond 5.3
+string(FIND ${LUA_DIR} lua-5.4 _is_lua_5_4)
+if(_is_lua_5_4)
+    find_package(Lua 5.4 EXACT REQUIRED) 
+else()
+    find_package(Lua REQUIRED) 
+endif()
```

Comment:
The `FindLua` module adds some necessary other libraries, like `-ld` to the lua target, which can be necessary for the `luac` library. I ran into an issue in my WSL build which had `hdf5` disabled (`hdf5` also adds `-ld`).